org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -430,6 +430,19 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
 			} else if (labelDetailsEnabled && $.getKind() == CompletionItemKind.Constructor && $.getLabelDetails() != null && $.getLabelDetails().getDetail() != null) {
 				filterText = newText.concat($.getLabelDetails().getDetail());
 			}
+			if (proposal.getKind() == CompletionProposal.METHOD_REF
```

Comment:
It's probbaly worth it to set `String.valueOf(proposal.getName())` to a variable given how many times you perform it. It'll simplify a lot of the other calls for readability.